lib/mongo/uri/srv_protocol.rb

```diff
@@ -78,12 +75,18 @@ def client_options
                                     "[#{VALID_TXT_OPTIONS.join(', ')}].".freeze
 
       MISTMATCHED_DOMAINNAME = "Parent domain name in SRV record result (%s) does not match " +
-                                 "hostname (%s)".freeze
+                                 "that of the hostname (%s)".freeze
+
+      FORMAT = 'mongodb+srv://[username:password@]host[/[database][?options]]'.freeze
```

Comment:
I think it should begin with `mongodb[+srv]` since the "+srv" is optional

Reply:
But for the SRV protocol, the +srv is required. There is this FORMAT variable for the mongodb scheme in the uri class: https://github.com/mongodb/mongo-ruby-driver/pull/898/files#diff-e7786d37b967fce59a4cd10176f3a85aR74
These are used in the error message for an InvalidURI error; at the time the error is instantiated, I know whether I need to pass in the FORMAT of the SRV protocol or the mongodb scheme.